### PR TITLE
Add basic MIME detection support

### PR DESCRIPTION
**Pull request by Uncurlhalo**
Adds super basic MIME detection support. Only bases detection off the first 512 Bytes at the moment.

### CODE REVIEW

File: upload.go [hunk 1]
```diff
@@ -12,9 +13,780 @@ import (
 	"strconv"
 	"strings"
 
+	"bitbucket.org/taruti/mimemagic"
 	"github.com/zenazn/goji/web"
 )
 
+var extensions = map[string]string{
```
Comment (andreimarcu) [inline upload.go]: Can we name this mimeToExtension and move it in the utils.go file?

File: upload.go [hunk 2]
```diff
@@ -119,6 +891,16 @@ func uploadHeaderProcess(r *http.Request, upReq *UploadRequest) {
 }
 
 func processUpload(upReq UploadRequest) (upload Upload, err error) {
+	// Get the mimetype of the upload via magic mime things
```
Comment (andreimarcu) [inline upload.go]: This should only happen if the extension is empty to begin with!

File: upload.go [hunk 3]
```diff
@@ -127,7 +909,7 @@ func processUpload(upReq UploadRequest) (upload Upload, err error) {
 	}
 
 	if len(extension) == 0 {
-		extension = "ext"
```
Comment (andreimarcu) [inline upload.go]: Same as above, you're basically replacing this extension = "ext" line with the mime detection.

File: upload.go (hunk 1 above)
Comment (Uncurlhalo) [inline upload.go]: Yes

File: upload.go [hunk 4]
```diff
@@ -127,7 +909,7 @@ func processUpload(upReq UploadRequest) (upload Upload, err error) {
...
+		extension = extensions[mimetype]
```
Comment (andreimarcu) [inline upload.go]: What if the mimetype isn't in the map? We want "ext" as fallback if the map doesn't contain it.

Comment (matthazinski) [inline upload.go]: Why "ext" and not something generic like "txt"?

Comment (andreimarcu) [inline upload.go]: Exactly that, it was meant to be generic vs "txt" being for text files. If you think there's something even better, I'm all ears.  I had considered .bin at a time but that implies the file is binary.

**Pull request merged**